### ISSUE TITLE
Feature request: Automatically add explicit markdown header IDs

### ISSUE THREAD

**Issue by corwintines** [ethereum/ethereum-org-website#10773]
### Is your feature request related to a problem? Please describe.

Resurfacing an old issue: https://github.com/ethereum/ethereum-org-website/issues/2304

Often when contributors add new markdown files, they forget to add explicit header IDs (e.g. see @wackerow's review of https://github.com/ethereum/ethereum-org-website/pull/2194).

### Describe the solution you'd like

We have a script to automatically add these explicit header IDs when they're not present:
https://github.com/ethereum/ethereum-org-website/blob/dev/src/scripts/generate-heading-ids.js

It can be executed from the project root, e.g.

$ yarn generate-heading-ids developers/tutorials/create-and-deploy-a-defi-app
Let's explore the potential to add this as a pre-commit hook (can we access the filename?) so that before any markdown files are committed, we run the generate-heading-ids script on the file.

### Describe alternatives you've considered

Add this as an instruction to the README, but none really. This would benefit most from being an automated check.

### Additional context

Why add add explicit header IDs: https://github.com/ethereum/ethereum-org-website/issues/272
TL;DR it's for anchor link consistency across translations

### Would you like to work on this issue?

- [ ] Yes
- [X] No

**Comment by wackerow** [comment]
`generate-heading-ids.js` was deprecated with Next.js migration. Closing this out for now; can re-open with updated requirements if still desired.
